Otus-DevOps-2018-02/ifqthenp_infra
# Homework 8. Terraform 2

# Completed Homework № 8

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Created terraform modularized configuration. The modules are `app`, `db`, and `vpc`
- :large_blue_diamond: Parameterized modules with custom variables
- :large_blue_diamond: Created Stage and Prod environments for modules configuration reuse
- :large_blue_diamond: Created file `storage-bucket.tf` with `storage-bucket` module configuration
- :large_orange_diamond: Added `backend.tf` config file to store terraform state on Google Cloud Storage platform
- :large_orange_diamond: Checked that state file is reachable in the bucket from different folders on the machine
- :large_orange_diamond: Added provisioning files to the modules `app` and `db`

## How to get project running

- `cd` into `terraform/stage` directory
- run commands `terraform init`, `terraform plan`, `terraform apply -auto-approve`, `terraform destroy -auto-approve`

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Type URL in browser address bar `http://35.197.199.73:9292` and login or sign up

## PR checklisk
